### PR TITLE
Remove webpack dir and setup demo module to use @labkey/build

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
With the port of the platform/webpack configuration files into a shared @labkey/build package, this PR now remove the separate copy of those files in this repository and updates the package.json files accordingly to use the shared versions.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/377
* https://github.com/LabKey/platform/pull/1654
* https://github.com/LabKey/inventory/pull/114
* https://github.com/LabKey/moduleEditor/pull/16
* https://github.com/LabKey/commonAssays/pull/251
* https://github.com/LabKey/tutorialModules/pull/29
* https://github.com/LabKey/provenance/pull/40

#### Changes
* Update module README files to reference updates doc information in @labkey/build
* Add package.json dependency for @labkey/build
* Update package.json scripts to use node_modules/@labkey/build configs
* Convert entryPoints generateViewHTML to standard generateLib

### CODE REVIEW

File: demo/README.md [hunk 1]
```diff
@@ -7,22 +7,25 @@ including keeping the standard LabKey header and header menus.
...
 - http://localhost:8080/labkey/home/demo-helloWorld.view?
 - http://localhost:8080/labkey/home/demo-todoList.view?
 
+Also, this module has an example of using the todoList page in a LabKey webpart.
+
 <a name="functionality"></a>
 # Functionality Overview
 The compilation and packaging of this module, including the NPM/webpack build, is done with the standard LabKey Gradle build. 
-Gradle will use the Node and NPM version defined in \<projectHome\>/gradle.properties to run the `build-prod` script defined in package.json.
+Gradle will use the Node and NPM version defined in `\<projectHome\>/gradle.properties` to run the `build-dev` script defined in package.json.
```
Comment (labkey-susanh) [inline demo/README.md]:
```suggestion
Gradle will use the Node and NPM version defined in `\<projectHome\>/gradle.properties` to run the `build-dev` script defined in `package.json`.
```

File: demo/README.md [hunk 2]
```diff
@@ -7,22 +7,25 @@ including keeping the standard LabKey header and header menus.
...
 Note: Gradle does not install Node/NPM globally. To run NPM commands outside of Gradle you will need to have Node/NPM installed 
-locally. For more information on what "scripts" the LabKey Gradle build expects from the module package.json file, 
+locally. For more information on what `scripts` the LabKey Gradle build expects from the module package.json file,
```
Comment (labkey-susanh) [inline demo/README.md]:
```suggestion
locally. For more information on what `scripts` the LabKey Gradle build expects from the module `package.json` file, 
```

File: demo/README.md [hunk 3]
```diff
@@ -7,22 +7,25 @@ including keeping the standard LabKey header and header menus.
...
-locally. For more information on what "scripts" the LabKey Gradle build expects from the module package.json file, 
+locally. For more information on what `scripts` the LabKey Gradle build expects from the module package.json file, 
 see the [Node.js Build Dependency] documentation page.
 
-The webpack production build will compile TypeScript and JavaScript files, as well as CSS and SCSS files, and bundle them independently 
-for each webpack entry point.  The bundled resources will be placed in the appropriate LabKey directory for web resources.  The production build will also 
-generate the necessary LabKey HTML files, including the containing elements for the React apps, as well as the necessary view.xml files 
-to make the bundled React and CSS files available to the appropriate LabKey React pages.
+The webpack build will compile TypeScript and JavaScript files, as well as CSS and SCSS files, and bundle them independently 
+for each module entry point, see example at `demo/src/client/entryPoints.js`.  The bundled resources will be placed in the
```
Comment (labkey-susanh) [inline demo/README.md]:
```suggestion
for each module entry point. See the example at `demo/src/client/entryPoints.js`.  The bundled resources will be placed in the 
```

File: demo/README.md [hunk 4]
```diff
@@ -7,22 +7,25 @@ including keeping the standard LabKey header and header menus.
...
+locally. For more information on what `scripts` the LabKey Gradle build expects from the module package.json file, 
 see the [Node.js Build Dependency] documentation page.
 
-The webpack production build will compile TypeScript and JavaScript files, as well as CSS and SCSS files, and bundle them independently 
-for each webpack entry point.  The bundled resources will be placed in the appropriate LabKey directory for web resources.  The production build will also 
-generate the necessary LabKey HTML files, including the containing elements for the React apps, as well as the necessary view.xml files 
-to make the bundled React and CSS files available to the appropriate LabKey React pages.
+The webpack build will compile TypeScript and JavaScript files, as well as CSS and SCSS files, and bundle them independently 
+for each module entry point, see example at `demo/src/client/entryPoints.js`.  The bundled resources will be placed in the 
+appropriate LabKey directory for web resources, see example at `resources/web/demo/gen/`.  The build will also
```
Comment (labkey-susanh) [inline demo/README.md]:
```suggestion
appropriate LabKey directory for web resources. See the example at `resources/web/demo/gen/`.  The build will also 
```

File: demo/README.md [hunk 5]
```diff
@@ -7,22 +7,25 @@ including keeping the standard LabKey header and header menus.
...
 
-The webpack production build will compile TypeScript and JavaScript files, as well as CSS and SCSS files, and bundle them independently 
-for each webpack entry point.  The bundled resources will be placed in the appropriate LabKey directory for web resources.  The production build will also 
-generate the necessary LabKey HTML files, including the containing elements for the React apps, as well as the necessary view.xml files 
-to make the bundled React and CSS files available to the appropriate LabKey React pages.
+The webpack build will compile TypeScript and JavaScript files, as well as CSS and SCSS files, and bundle them independently 
+for each module entry point, see example at `demo/src/client/entryPoints.js`.  The bundled resources will be placed in the 
+appropriate LabKey directory for web resources, see example at `resources/web/demo/gen/`.  The build will also 
+generate the necessary LabKey HTML files, including the containing elements for the React apps, as well as the necessary 
+`view.xml` files to make the bundled React and CSS files available to the appropriate LabKey React pages, see example at `resources/views/`.
```
Comment (labkey-susanh) [inline demo/README.md]:
```suggestion
`view.xml` files to make the bundled React and CSS files available to the appropriate LabKey React pages. See the example at `resources/views/`.
```

File: demo/src/client/ToDoListPage/ToDoList.tsx [hunk 6]
```diff
@@ -38,7 +38,7 @@ export const ToDoList: FC<ToDoListProps> = memo(({ items, onItemClick }) => {
             <div className="panel-body">
                 <p>To-Do:</p>
 
-                {!hasItems && <p>Add a To-Do to your list.</p>}
+                {!hasItems && <p className={'todo-empty'}>Add a To-Do to your list.</p>}
```
Comment (labkey-susanh) [inline demo/src/client/ToDoListPage/ToDoList.tsx]:
```suggestion
                {!hasItems && <p className="todo-empty">Add a To-Do to your list.</p>}
```
If you use the double quotes for className, IntelliJ can actually navigate to the class in the scss file, which is handy.

File: demo/resources/views/demoWebpart.view.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<view xmlns="http://labkey.org/data/xml/view" frame="none" title="To-Do List Webpart">
```
Comment (labkey-susanh) [inline demo/resources/views/demoWebpart.view.xml:1]: This is a file generated by webpack, right?  I wonder if folks will take this as a sign that the `.view.xml` files should be checked in.

**Review by labkey-susanh — APPROVED** [review]
Just some minor suggestions and one question.

File: demo/README.md (hunk 1 above)
Comment (cnathe) [reply demo/README.md]: done

File: demo/README.md (hunk 2 above)
Comment (cnathe) [reply demo/README.md]: done

File: demo/README.md (hunk 3 above)
Comment (cnathe) [reply demo/README.md]: done

File: demo/README.md (hunk 4 above)
Comment (cnathe) [reply demo/README.md]: done

File: demo/README.md (hunk 5 above)
Comment (cnathe) [reply demo/README.md]: done

File: demo/src/client/ToDoListPage/ToDoList.tsx (hunk 6 above)
Comment (cnathe) [reply demo/src/client/ToDoListPage/ToDoList.tsx]: Ah, I didn't realize that. That's very helpful. Done.

File: demo/resources/views/demoWebpart.view.xml (hunk 7 above)
Comment (cnathe) [reply demo/resources/views/demoWebpart.view.xml:1]: Nope, this webpart one is not webpack generated. The entryPoints.js has a 'demoWebpart' entry which has `generateLib: true`. These related html/view.xml/webpart/xml files are added manually in this case for the webpart scenario.

**Review by labkey-martyp — APPROVED** [review]

**Pull request merged**